src/main/app/auth/service/get-auth-token.ts

```diff
@@ -0,0 +1,30 @@
+import { Logger } from '@hmcts/nodejs-logging';
+import Axios from 'axios';
+import config from 'config';
+import OTP from 'otp';
+
+const logger = Logger.getLogger('service-auth-token');
+let token;
+
+const getTokenFromApi = () => {
+  logger.info('Refreshing service auth token');
+
+  const url: string = config.get('services.authProvider.url');
+  const microservice: string = config.get('services.authProvider.microservice');
+  const secret: string = config.get('services.authProvider.secret');
+  const oneTimePassword = new OTP({ secret }).totp();
+  const body = { microservice, oneTimePassword };
+
+  Axios.post(url, body)
+    .then(response => (token = response.data))
+    .catch(err => logger.error(err));
+};
+
+export const initAuthToken = (): void => {
+  getTokenFromApi();
+  setInterval(getTokenFromApi, 1000 * 60 * 60);
+};
+
+export const getAuthToken = (): string => {
+  return token;
+};
```

Comment:
is an auth token for the service and not the user?

Reply:
there are two, I should rename this method to be clearer: getServiceAuthToken